src/main/java/beans/EvaluateXPathResponseBean.java

```diff
@@ -31,17 +35,31 @@
     private String status;
     private String contentType;
 
+    private String trace;
+
     //Jackson requires the default constructor
     public EvaluateXPathResponseBean(){}
 
-    public EvaluateXPathResponseBean(EvaluationContext evaluationContext, String xpath) throws XPathSyntaxException {
+    public EvaluateXPathResponseBean(EvaluationContext evaluationContext, String xpath, String debugTraceLevel) throws XPathSyntaxException {
         status = Constants.ANSWER_RESPONSE_STATUS_POSITIVE;
         ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
         KXmlSerializer serializer = new KXmlSerializer();
 
+        EvaluationContext subcontext = evaluationContext.spawnWithCleanLifecycle();
```

Comment:
Why do we need to spawn a new `subContext` here? Just being a good citizen?

Reply:
The  debugger we set after this point is set on an eval context, so we have to spin up a new context to prevent leaking anything from outside of this method/process